components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -736,6 +736,8 @@ interface JsonRpcService {
   SetNetwork(string chain_id, CoinType coin) => (bool success);
   GetNetwork(CoinType coin) => (NetworkInfo network);
   GetAllNetworks(CoinType coin) => (array<NetworkInfo> networks);
+  GetCustomNetworks(CoinType coin) => (array<string> chain_ids);
+  GetKnownNetworks(CoinType coin) => (array<string> chain_ids);
```

Comment:
The return values I see in the issue is `array<NetworkInfo> networks` instead of array of strings, is the requirement updated? cc @kylehickinson @StephenHeaps to double confirm what return type would be preferred.

Reply:
I've made that in the same way as for settings webui. 
And to avoid confusion some chain_id could be both in "known" and "custom" list
"known" and "custom" is resettable
"known" is not removable

I was tempted to put these into NetworkInfo as bool fields but it appeared quite complicated to cover all scenarios(like persisting and loading from prefs, etc)
I would consider these two arrays as external flags. Maybe worth refactoring at some point when iOS and Android are done.

So do we still need array<NetworkInfo> here?

Reply:
Just the `chain_id` would be fine if that reduces complexity. 

On iOS we are already fetching all networks on the screens where these functions would be used, so we can filter to for example just the custom networks based on the `chain_id` provided from this `GetCustomNetworks` function.